### PR TITLE
Android: Handle Cookie login refresh in Mobile SDK 

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: android/src/main/java/com/liferay/mobile/android/auth/basic/CookieAuthentication.java [hunk 1]
```diff
@@ -17,20 +17,35 @@
...
 
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 /**
  * @author Javier Gamarra
  */
 public class CookieAuthentication extends BasicAuthentication {
 
+	public final static int COOKIE_EXPIRATION_TIME = 15 * 60;
```
Comment (jmnavarro) [inline android/src/main/java/com/liferay/mobile/android/auth/basic/CookieAuthentication.java]: why public?

Comment (victorg1991) [reply android/src/main/java/com/liferay/mobile/android/auth/basic/CookieAuthentication.java]: Mainly to use it from Screens and not define the same constant there

File: android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+                if (callback == null) {
+
+                    try {
+                        Session cookieSession = CookieSignIn.signIn(session, authenticator);
+                        request.setAuthentication(cookieSession.getAuthentication());
+
+                        return cookieSession;
+                    } catch (Exception e) {
+                        throw e;
```
Comment (nhpatt) [inline android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: why?

File: android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        final Session session = new SessionImpl(request.getURL(), request.getAuthentication());
+
+        if ((request.getAuthentication() instanceof CookieAuthentication)
+                && ((CookieAuthentication) request.getAuthentication()).shouldHandleExpiration()) {
+
+            CookieAuthentication authentication = (CookieAuthentication) request.getAuthentication();
+
+            lock.lock();
+
+            if (shouldRefreshCookie(authentication.getLastCookieRefresh(),
```
Comment (nhpatt) [inline android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: Can you extract both sides of the if to a method?

File: android/src/main/java/com/liferay/mobile/android/auth/CookieSignIn.java [hunk 4]
```diff
@@ -66,7 +66,7 @@ public static Session signIn(Session session, Authenticator authenticator) throw
 
 		return parseResponse(
 			response, session.getServer(), cookieSignIn.cookieManager,
-			cookieSignIn.username, cookieSignIn.password);
+				(CookieAuthentication) session.getAuthentication());
```
Comment (nhpatt) [inline android/src/main/java/com/liferay/mobile/android/auth/CookieSignIn.java]: extract to method that casts

File: android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java (hunk 2 above)
Comment (victorg1991) [reply android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: I don't want to break the current flow, so the exception is handled in the same place than before

Comment (nhpatt) [reply android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: But you just rethrow it, why not remove that?

Comment (victorg1991) [reply android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: I need the finally clause to unlock the Lock

File: android/src/main/java/com/liferay/mobile/android/auth/CookieSignIn.java (hunk 4 above)
Comment (victorg1991) [reply android/src/main/java/com/liferay/mobile/android/auth/CookieSignIn.java]: Done! :D

File: android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java (hunk 3 above)
Comment (victorg1991) [reply android/src/main/java/com/liferay/mobile/android/auth/CookieExpirationHandler.java]: Done! :D

**Comment by allanragec** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by allanragec** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by allanragec** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by allanragec** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by allanragec** [comment]
Thank you, pull request merged! [See changes here.](https://github.com/liferay/liferay-mobile-sdk/compare/d5b91ad4~3...d5b91ad4) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request merged**